src/main/java/service/AnimalSerializer.java

```diff
@@ -0,0 +1,46 @@
+package service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import model.Animal;
+
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AnimalSerializer {
+    private static ObjectMapper mapper;
+    private static String path = "src/main/resources/animalOut";
+
+    public AnimalSerializer(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    public static void serialize(List<Animal> petShelter) {
+        try {
+            mapper.writeValue(new File(path + ".json"), petShelter);
+        } catch (IOException e) {
+            System.out.println("cannot create file !");
+            e.printStackTrace();
+        }
+    }
+
+    public static List<Animal> deserialize() {
+        List<Animal> animals = new ArrayList<>();
+        try {
+            File animalList = new File("src/main/resources/animalOut.json");
+            animals = mapper.readValue(animalList, List.class);
```

Comment:
Trying to get a pet after "Do you want to took any pets? Yes / No" = Yes you'll recieve a 
class java.util.LinkedHashMap cannot be cast to class model.Animal
cause in this line you are reading a List.... List of what? java doesnt know that JSON objects in your files should be mapped to Animal object
could be solved by 
animals = List.of(mapper.readValue(animalList, Animal[].class));